### ISSUE TITLE
Using weave with rancher ?

### ISSUE THREAD

**Issue by fbrbovic** [rancher/rancher#4752]
Was anyone successful in integrating weave (https://www.weave.works/products/weave-net/) with rancher, any instructions or tutorials on how to do it anywhere ?
Also any plans from rancher of supporting it in future?

**Comment by andrewmclagan** [comment]
Why not use compose instead?

**Comment by fbrbovic** [comment]
What do you mean by using compose ?

**Comment by deniseschannon** [comment]
@fbrbovic I'm assuming that @andrewmclagan meant why wouldn't you use rancher-compose.
http://docs.rancher.com/rancher/latest/en/rancher-compose/

I know very little about weave, but at this time we have no plans to incorporate integrating it within rancher.

**Comment by mishak87** [comment]
@deniseschannon Since [0.15 Weaveworks Scope](https://www.weave.works/scope-0-15-adds-search-enhanced-kubernetes-visualization/) has support for meta information from kubernetes. It would be nice to have Rancher plugin providing similar insight.

I thought about writing simple plugin but have no time for it.

**Comment by fbrbovic** [comment]
@deniseschannon @andrewmclagan  I know what Rancher-Compose is, I didn't know how using that instead of weave-net would work as a replacement.  Weave-net  provides networking enhancement to the overlay networking of docker and in this case would be of rancher. 

I wish rancher would allocate more development time on networking part of it, I love what rancher is doing but I think it is lacking on the networking front. 

I know that this should probably be a separate issue but here is my two big wishlist items for rancher networking

1. Make ipsec and tunneling optional for private networks ( some of us host servers on private vlans) and this would impre performance. 
2. Allow for some kind of private sub-networks that containers can live on or have some kind of internal firewall which would allow us to restrict traffic between stacks or containers.  Projects like [weave-net ](https://www.weave.works/products/weave-net/) and [callico ](https://www.projectcalico.org/learn/) provide such capabilities. 
I did mention this in one other issue https://github.com/rancher/rancher/issues/3895

**Comment by deniseschannon** [comment]
@fbrbovic Another issue might be helpful. Just to let you know, we've also hired @leodotcloud who will be helping us advance our networking in Rancher and we'll be looking at some of the things you're asking about.

**Comment by fbrbovic** [comment]
That's great news @deniseschannon :)

**Comment by xeor** [comment]
We are testing out rancher as well but are getting blocked by the lack of network support. Currently, it seams that every container can talk to everyone. Which is a huge issue if we want to use rancher for anything.. I don't even want that @ home.

**Comment by leodotcloud** [comment]
@fbrbovic : Could you please share your thoughts as to why you are inclined towards using weave driver?
@xeor : Can you please share your use cases in more detail? That would help me during the design phase.

**Comment by xeor** [comment]
@leodotcloud I think a generic solution supporting what docker supports is a good way of doing this.
The UI part can be problematic, but maybe it can be done in the same way as volumes?

Isolation of services / apps is probably the most urgent problem, separating it so applications can't talk to each others. It should be possible to create networks per applications, maybe even define rules so that a container can only talk to specific networks... Basically, everything docker is starting to support native. Or with plugins such as Weave.

Network drivers such as ipvlan / macvlan would also be needed.

To sum up, Rancher needs support for docker network drivers and docker native network :+1:

**Comment by jhmartin** [comment]
+1 to application isolation.  Running a swarm tends to give up the use of some of the security provided by AWS security groups; docker stack-specific networks regain some of that capability. I'd like app A's webserver to be able to talk to its redis, but not app B's database server.

**Comment by InAnimaTe** [comment]
I don't know a ton about CNI or the modern docker network drivers (Calico, Weave, etc..) but it's sounding more and more like the modularity being introduced with 1.2 is going to solve a lot of these asks about separation and increased security between services/stacks. 

My core interest in this is similar to @xeor whereas my company is trying to adopt Rancher as our infrastructure orcheatration framework for our SaaS based platform. In this regard, we're obviously interested in the networking logistics available when scheduling containers across an array of hosts. With the flexibility platforms like GCE and AWS provide with things like VPC's, some of those similar concepts really need to be made available in Rancher, whether that's through merely supporting Docker network drivers flawlessly or by other means.

Driving down a bit deeper, we're really interested in secure networking between services in a stack. To us, a stack is a great logical object whereas separation can easily be defined between actual applications, which is usually preferred. If someone enters app x, they shouldn't be able to reach app y.

I'm digging a bit deeper into what Calico or Weave can provide me for now, but I worry about how I might still be able to utilize many Rancher network services DNS, LB, or Metadata. I definitely think taking time to ensure Rancher network services can be made to work no matter the driver used would be exceptionally appreciated by many users.

**Comment by leodotcloud** [comment]
We have a lot of infrastructure updates coming up in the next release and as part of our on going efforts we are planning to add support for more networking plugins like Calico, Weave.

**Comment by meggarr** [comment]
Hi, I am trying Rancher as well. And I'd like to know if the Weave Net support is in the coming releases? Thanks!

**Comment by qLb** [comment]
@meggarr @leodotcloud any news on the weave network plugin integrations? i found something here https://github.com/rancher/community-catalog/tree/master/kubernetes-templates/weave-cloud but could already be outdated because weave.works propose using this `kubectl apply -f https://git.io/weave-kube` for daemon set configuration on k8s.

ps. if anyone's interested, this is how you enable traffic encryption for weave overlay network on k8s the native way: https://github.com/weaveworks-experiments/weave-kube/issues/38#issuecomment-253855874

Cheers!

**Comment by dbason** [comment]
@leodotcloud is there somewhere we can subscribe to the ongoing work for this?  We're also very keen to see WeaveNet as a supported CNI driver for k8s on Rancher.

FastPath encryption and policy based networking are both features we really like about WeaveNet.

**Comment by errordeveloper** [comment]
@leodotcloud I'd be happy to help here, if you need more info on Weave Net.

cc @mindfulmonk

**Comment by deniseschannon** [comment]
With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.
